orte/mca/rml/ofi/rml_ofi_component.c

```diff
@@ -0,0 +1,1091 @@
+            }
+
+            /**
+            * Enable the endpoint for communication
+            * This commits the bind operations.
+            */
+            ret = fi_enable(orte_rml_ofi.ofi_conduits[cur_conduit].ep);
+            if (0 != ret) {
+                opal_output_verbose(1, orte_rml_base_framework.framework_output,
+                                    "%s:%d: fi_enable failed: %s\n",
+                                    __FILE__, __LINE__, fi_strerror(-ret));
+                free_conduit_resources(cur_conduit);
+                /* abort this current transport, but check if next transport can be opened */
+                continue;                               
+            }
+            opal_output_verbose(10,orte_rml_base_framework.framework_output,
+                            "%s:%d ep enabled for conduit_id - %d ",__FILE__,__LINE__,cur_conduit);
+
+
+            /**
+            * Get our address and publish it with modex.
+            **/
+            orte_rml_ofi.ofi_conduits[cur_conduit].epnamelen = sizeof (orte_rml_ofi.ofi_conduits[cur_conduit].ep_name);
+            ret = fi_getname((fid_t)orte_rml_ofi.ofi_conduits[cur_conduit].ep,
+                            &orte_rml_ofi.ofi_conduits[cur_conduit].ep_name[0],
+                            &orte_rml_ofi.ofi_conduits[cur_conduit].epnamelen);
+            if (ret) {
+                        opal_output_verbose(1, orte_rml_base_framework.framework_output,
+                                    "%s:%d: fi_getname failed: %s\n",
+                                    __FILE__, __LINE__, fi_strerror(-ret));
+                free_conduit_resources(cur_conduit);
+                /* abort this current transport, but check if next transport can be opened */
+                  continue;
+            }
+
+            switch ( orte_rml_ofi.ofi_conduits[cur_conduit].fabric_info->addr_format)
+            {
+                case  FI_SOCKADDR_IN :
+                    opal_output_verbose(1,orte_rml_base_framework.framework_output,
+                            "%s:%d In FI_SOCKADDR_IN.  ",__FILE__,__LINE__);
+                    /*  Address is of type sockaddr_in (IPv4) */
+                    opal_output_verbose(1,orte_rml_base_framework.framework_output,
+                            "%s sending Opal modex string for conduit_it %d, epnamelen = %d  ",
+                            ORTE_NAME_PRINT(ORTE_PROC_MY_NAME),cur_conduit,orte_rml_ofi.ofi_conduits[cur_conduit].epnamelen);
+                    /*[debug] - print the sockaddr - port and s_addr */
+                    struct sockaddr_in* ep_sockaddr = (struct sockaddr_in*)orte_rml_ofi.ofi_conduits[cur_conduit].ep_name;
+                    opal_output_verbose(1,orte_rml_base_framework.framework_output,
+                            "%s port = 0x%x, InternetAddr = 0x%x  ",
+                            ORTE_NAME_PRINT(ORTE_PROC_MY_NAME),ep_sockaddr->sin_port,ep_sockaddr->sin_addr.s_addr);
```

Comment:
You should use `inet_ntoa(ep_sockaddr->sin_addr)` to print a string of the IPv4 address, and use `ntohs(ep_sockaddr->sin_port)` to print the host representation of the port (otherwise, it'll be a network byte order).